src/14-curated-discovery/01-artworks.ts

```diff
@@ -0,0 +1,286 @@
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: false,
+            vectorizePropertyName: true,
+          },
+        },
+      },
+      {
+        name: "tags",
+        dataType: ["text[]"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: false,
+            vectorizePropertyName: true,
+          },
+        },
+      },
+      {
+        name: "additionalInformation",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: false,
+            vectorizePropertyName: true,
+          },
+        },
+      },
+      {
+        name: "imageUrl",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+          },
+        },
+      },
+      {
+        name: "imageDescription",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: false,
+            vectorizePropertyName: true,
+          },
+        },
+      },
+      {
+        name: "artistID",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+          },
+        },
+      },
+      {
+        name: "partnerID",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+          },
+        },
+      },
+    ],
+  }
+
+  const classResult = await client.schema
+    .classCreator()
+    .withClass(classWithProps)
+    .do()
+
+  console.log(JSON.stringify(classResult, null, 2))
+}
+
+/**
+ * Insert artworks into Weaviate
+ *
+ * Also assigns a deterministic UUID to each artwork,
+ * based on its Gravity ID
+ */
+async function insertArtworks(
+  artworks: GravityArtwork[],
+  batchSize: number = BATCH_SIZE
+) {
+  console.log(`Inserting artwork: ${artworks.length}`)
+
+  const batches = _.chunk(artworks, batchSize)
+  console.log(`Inserting ${batches.length} batches`)
+
+  for (const artworkBatch of batches) {
+    let batcher = client.batch.objectsBatcher()
+    batcher = batcher.withObjects(
+      ...artworkBatch.map((artwork) => {
+        return {
+          class: CLASS_NAME,
+          properties: {
+            internalID: artwork.id,
+            slug: artwork.slug,
+            title: artwork.title,
+            date: artwork.date,
+            rarity: artwork.rarity,
+            medium: artwork.medium,
+            materials: artwork.materials,
+            dimensions: artwork.dimensions,
+            price: artwork.price,
+            listPriceAmount: artwork.list_price_amount,
+            listPriceCurrency: artwork.list_price_currency,
+            artworkLocation: artwork.artwork_location,
+            categories: artwork.categories,
+            tags: artwork.tags,
+            additionalInformation: artwork.additional_information,
+            imageUrl: artwork.image_url,
+            imageDescription: null, // TODO: add computer vision step?
```

Comment:
Two likely follow-ups here:

- a separate class that will embed artworks via multi2vec and make use of the image data specified on L271
- an update to this class that will persist a computer vision description of the artwork image into the field on L272